aiida/restapi/run_api.py

```diff
@@ -37,18 +39,53 @@ def run_api(flask_app=api_classes.App, flask_api=api_classes.AiidaApi, **kwargs)
     :param debug: enable debugging
     :param wsgi_profile: use WSGI profiler middleware for finding bottlenecks in web application
     :param hookup: If true, hook up application to built-in server - else just return it
-    """
-    # pylint: disable=too-many-locals
 
-    # Unpack parameters
+    :returns: tuple (app, api) if hookup==False or runs app if hookup==True
+    """
+    hookup = kwargs.pop('hookup', CLI_DEFAULTS['HOOKUP_APP'])
     hostname = kwargs.pop('hostname', CLI_DEFAULTS['HOST_NAME'])
     port = kwargs.pop('port', CLI_DEFAULTS['PORT'])
-    config = kwargs.pop('config', CLI_DEFAULTS['CONFIG_DIR'])
+    debug = kwargs.pop('debug', APP_CONFIG['DEBUG'])
+
+    app, api = configure_api(flask_app, flask_api, **kwargs)
+
+    if hookup:
+        # Run app through built-in werkzeug server
+        print(' * REST API running on http://{}:{}{}'.format(hostname, port, API_CONFIG['PREFIX']))
+        api.app.run(debug=debug, host=hostname, port=int(port), threaded=True)
 
+    else:
+        warnings.warn(  # pylint: disable=no-member
+            'Using `run_api` with `hookup==False` is deprecated since `v1.2.1` and will stop working in `v2.0.0`. '
+            'Use `configure_api` instead.', AiidaDeprecationWarning
+        )
+
+        # Return the app & api without specifying port/host to be handled by an external server (e.g. apache).
+        # Some of the user-defined configuration of the app is ineffective (only affects built-in server).
+        return (app, api)
+
+
+def configure_api(flask_app=api_classes.App, flask_api=api_classes.AiidaApi, **kwargs):
```

Comment:
if this is supposed to be used by users of the package, may want to add it to the `__all__` at the top of the file